### PR TITLE
Incykiosk 108

### PR DESCRIPTION
**Pull request by andersnylund**
Enables to give the category and place as URL parameters. Simplifies the checking of the next question.

### CODE REVIEW

File: src/actions/choiceActions.js [hunk 1]
```diff
@@ -17,9 +16,15 @@ export const getAllChoicesAction = (questions, langId) => {
 
         try {
             // gets the choices concurrently
-            const choices = await Promise.all(questions.map(question => service.getChoices(question.id, langId)));
+            const choices = await Promise.all(questions.map(async question => {
```
Comment (VonLatvala) [inline src/actions/choiceActions.js]: geez, a real mixin' async and promises :D

File: src/service.js [hunk 2]
```diff
@@ -74,20 +92,34 @@ const getLanguages = () => {
...
-const getCategory = (langId) => {
-    return getUrl(categoryUrl, {
+const getCategory = async (langId) => {
+    const categories = await getUrl(categoryUrl, {
         headers: {
             "Accept-Language": (langId + ';q=1'),
         },
     });
+    const selectedCategory = categoryName;
+    const foundCategory = categories.find(category => category.name === selectedCategory);
```
Comment (VonLatvala) [inline src/service.js]: _should_ typecheck and throw/log error if incorrect

File: src/service.js [hunk 3]
```diff
@@ -74,20 +92,34 @@ const getLanguages = () => {
 };
 
 // TODO: headers (the language) should be somehow automatically added to each request
-const getCategory = (langId) => {
-    return getUrl(categoryUrl, {
+const getCategory = async (langId) => {
+    const categories = await getUrl(categoryUrl, {
```
Comment (VonLatvala) [inline src/service.js]: what happens on 4XX 5XX HTTP error codes?

Comment (VonLatvala) [reply src/service.js]: nevermind, getUrl does a `console.error` on error

File: src/service.js [hunk 4]
```diff
@@ -74,20 +92,34 @@ const getLanguages = () => {
...
+        return foundCategory;
+    } else {
+        return categories[0];
+    }
 };
 
-const getPlace = (langId) => {
-    return getUrl(placeUrl, {
+const getPlace = async (langId) => {
+    const places = await getUrl(placeUrl, {
```
Comment (VonLatvala) [inline src/service.js]: _should_ typecheck and throw/log error if incorrect

**Comment by andersnylund** [comment]
Forgot to implement logic that handles skipping questions

**Pull request merged**